### PR TITLE
docs: :building_construction: diagram of internal code for `extract_resource_properties()`

### PR DESCRIPTION
**Pull request by lwjohnst86**
## Description

I split these out from the diagram @signekb started on, after moving out the extraction of properties function (see #602).

## Reviewer Focus

This PR needs an in-depth review.

### CODE REVIEW

File: docs/design/images/core-python-functions/extract-resource-properties.puml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+  state "verify_is_supported_format()" as vsf
+  vsf : - Required arg: path
+  vsf : - Output: path or Error
+  vif --> vsf : path object
+
+  state "extract_properties_from_file()" as epff
+  epff : - Required arg: path
+  epff : - Output: JSON
+  vsf --> epff : JSON object
```
Comment (martonvago) [inline docs/design/images/core-python-functions/extract-resource-properties.puml:50]:
```suggestion
  vsf --> epff : path object
```
Right?

**Review by martonvago — CHANGES_REQUESTED** [review]
Amazing, makes sense to me 😁

File: docs/design/images/core-python-functions/extract-resource-properties.puml (hunk 1 above)
Comment (signekb) [reply docs/design/images/core-python-functions/extract-resource-properties.puml:50]: Would make sense :)

**Review by signekb — COMMENTED** [review]
It also makes sense to me! I agree with @martonvago 's comment :)

**Pull request merged**